### DISCUSSION TITLE
Preview show final website

### DISCUSSION THREAD

**Discussion by tobiasBora** [decaporg/decap-cms#7178] [Q&A]
This project is awesome! But I'm curious, is it possible to preview the final website on the right preview instead of an approximation? I saw that this library provides custom functions to customize the preview, but first I don't know how to get the content of other files (e.g. to create a menu or print a preview of a list of articles), but I also don't feel like re-writting my website twice, as it not only doubles the amount of work, but it is also very error prone.

Of course, I do understand that it might not be easy on complex websites like Hugo that need a compilation process (even if it might be possible to use WASM and compile it in the browser ^^), but I guess it might be easier to do for Next.js/Nuxt.js/sveltekit websites. Any advice on how to do that?

**Comment by mnordhaus** [comment]
Hi Tobias,

you might what to look to this article: https://github.com/StaticJsCMS/static-cms/discussions/469

Currently the Preview topic has not too much urgency for me, but I see it as important as well. Especially the aspect of writing code twice. I think a smart concept of using html attributes and a good javascript magic should do the trick eventually. But in the linked discussions are already great though. FYI: StaticCMS is a fork from Netlify. Basically a sister project - so the resources are quite compatible.

Please keep me in the look with your findings. I am happy to chime in and discuss things with you. But I have not enough time to actively develop that topic right now.

Best,

Markus

**Reply by tobiasBora** [reply]
Thanks a lot for both resources, they look great! I will try when I can, but for now proper support for MDX with nested components is forbidding me from using this great tool as reported in https://github.com/decaporg/decap-cms/issues/7180

**Comment by mnordhaus** [comment]
Ah and see also: https://github.com/decaporg/decap-cms/issues/2183#issuecomment-997373169
